src/main/java/seedu/address/ui/WidgetViewBox.java

```diff
@@ -63,22 +64,23 @@
     public WidgetViewBox(ObservableList<Client> clientObservableList) {
         super(FXML);
         this.clientObservableList = clientObservableList;
+        clientListSize = clientObservableList.size();
         clientObservableList.addListener((ListChangeListener<Client>) c -> {
             if (c.next()) {
-                if (c.wasUpdated() || c.wasReplaced()) {
+                if (clientObservableList.size() == clientListSize && !c.wasPermutated()) {
                     if (displayedClientIndex != -1) {
                         updateClientDisplay(clientObservableList.get(displayedClientIndex));
-                    }
-                } else {
-                    for (int i = 0; i < clientObservableList.size(); i++) {
-                        Client client = clientObservableList.get(i);
-                        if (client.isSameClient(displayedClient)) {
-                            displayedClient = client;
-                            displayedClientIndex = i;
-                            return;
+                    } else {
+                        for (int i = 0; i < clientObservableList.size(); i++) {
+                            Client client = clientObservableList.get(i);
+                            if (client.isSameClient(displayedClient)) {
+                                displayedClient = client;
+                                displayedClientIndex = i;
+                                return;
+                            }
                         }
```

Comment:
This else block should belong to the other if statement (if (clientObservableList.size() == clientListSize && !c.wasPermutated())), and not the inner one. If not when you remove the client you're currently viewing it won't go back to default view

Reply:
ok changed